march_hardware/src/PDOmap.cpp

```diff
@@ -35,70 +35,106 @@ std::map<enum IMCObjectName, int> PDOmap::map(int slaveIndex, enum dataDirection
   {
     reg = 0x1A00;
     SMAddress = 0x1C13;
-  }
-  else if (direction == dataDirection::mosi)
-  {
-    reg = 0x1600;
-    SMAddress = 0x1C12;
-  }
-  else
-  {
-    ROS_ERROR("Invalid dataDirection argument");
-  }
-  // Clear SyncManager Object
-  sdo_bit8(slaveIndex, SMAddress, 0, 0);
-  int startReg = reg;
-  int lastFilledReg = reg;
-  int sizeleft = this->bitsPerReg;
-  int counter = 0;
-  int byteOffset = 0;
-  while (this->sortedPDOObjects.size() > 0)
-  {
-    // Check if register is still empty
-    if (sizeleft == this->bitsPerReg)
+
+    sdo_bit8(slaveIndex, SMAddress, 0, 0);
+    int sizeleft = this->bitsPerReg;
+    int counter = 0;
+    int byteOffset = 0;
+    int count = 0;
+    int startReg = reg;
+    int lastFilledReg = reg;
+
+    while (this->sortedPDOObjects.size() > 0)
     {
-      sdo_bit32(slaveIndex, reg, 0, 0);
+      // Check if register is still empty
+      if (sizeleft == this->bitsPerReg)
+      {
+        sdo_bit32(slaveIndex, reg, 0, 0);
+      }
+      // Get next object (from end, because sorted from small to large)
+      std::pair<IMCObjectName, IMCObject> nextObject = this->sortedPDOObjects.back();
+      this->sortedPDOObjects.pop_back();
+      // Add next object to map
+      counter++;
+      sdo_bit32(slaveIndex, reg, counter,
+                this->combineAddressLength(nextObject.second.address, nextObject.second.length));
+      this->byteOffsets[nextObject.first] = byteOffset;
+      byteOffset += nextObject.second.length / 8;
+      sizeleft -= nextObject.second.length;
+      // Check if this was the last object of the list
+      if (this->sortedPDOObjects.size() == 0)
+      {
+        sdo_bit32(slaveIndex, reg, 0, counter);
+        lastFilledReg = reg;
+        reg++;
+      }
+      // else, check if register is full
+      else if (sizeleft <= 0)
+      {
+        sdo_bit32(slaveIndex, reg, 0, counter);
+        reg++;
+        counter = 0;
+        sizeleft = this->bitsPerReg;
+      }
```

Comment:
What are these changes? Just indentation?

Reply:
Nope it seems it still uses hardcoded pdo mapping. Please go back to the dynamic pdo mapping as defined in develop and check if that works.

Reply:
This explains why torque mode even works despite not adding it as a mosi pdo object in the imc class, btw.